projects/arlas-toolkit/src/lib/components/config-manager/action-modal/action-modal.component.ts

```diff
@@ -38,100 +43,172 @@ export class ActionModalComponent {
+      .pipe(mergeMap((p: DataWithLinks) =>
+        this.persistenceService.create('i18n', newI18nName, p.doc_value, [], [],
+          this.getOptionsSetOrg(p.doc_organization))
+      ));
+  }
+
+  private duplicateTour$(i18nId: string, lg: string, newConfigName: string, org?: string): Observable<DataWithLinks> {
+    const newTourName = newConfigName.concat('_tour_' + lg);
+    return this.persistenceService.get(i18nId, this.getOptionsSetOrg(org))
+      .pipe(mergeMap((p: DataWithLinks) =>
+        this.persistenceService.create('tour', newTourName, p.doc_value, [], [],
+          this.getOptionsSetOrg(p.doc_organization))
+      ));
+  }
+
+  public rename(newName: string, config: Config) {
+    const options = this.getOptionsSetOrg(config.org);
+    this.persistenceService.get(config.id, options)
+      .pipe(
+        catchError((err) => {
+          this.errorService.closeAll().afterAllClosed.pipe(take(1))
+            .subscribe(() =>
+              this.errorService.emitAuthorisationError(new AuthorisationOnActionError(err.status, 'rename_dashboard'), false));
+          return of(err);
+        })
+      )
       .subscribe(
-        data => {
-          this.errorMessage = '';
-          this.dialogRef.close(data.id);
-        },
-        error => this.raiseError(error));
+        (currentConfig: DataWithLinks) => {
+          const arlasConfig = this.configurationService.parse(currentConfig.doc_value);
+          if (!!arlasConfig) {
+            if (this.configurationService.hasPreview(arlasConfig)) {
+              const previewId = this.configurationService.getPreview(arlasConfig);
+              this.persistenceService.renameResource(previewId, newName + '_preview', options);
+            }
+            if (this.configurationService.hasI18n(arlasConfig)) {
+              const i18nIds = this.configurationService.getI18n(arlasConfig);
+              Object.keys(i18nIds).forEach(lg => {
+                this.persistenceService.renameResource(i18nIds[lg], newName + '_i18n_' + lg, options);
+              });
+            }
+            if (this.configurationService.hasTours(arlasConfig)) {
+              const toursIds = this.configurationService.getTours(arlasConfig);
+              Object.keys(toursIds).forEach(lg => {
+                this.persistenceService.renameResource(toursIds[lg], newName + '_i18n_' + lg, options);
+              });
+            }
+          }
+          this.persistenceService.rename(config.id, newName, this.getOptionsSetOrg(config.org))
+            .subscribe({
+              error: error => this.raiseError(error),
+              next: () => {
+                this.dialogRef.close(this.action);
+              }
+            });
+        });
   }
 
   public closeShare(event: [boolean, any]) {
     // update share is successful, close dialog
     if (event[0]) {
-      this.dialogRef.close();
+      this.dialogRef.close(event[1]);
```

Comment:
The test is on `event[0]` and you use `event[1]`

Reply:
yes, I only check if I should close with event[0] which is a boolean and put the content (event[1]) in the close action